network_diffusion/tpn/functions.py

```diff
@@ -0,0 +1,50 @@
+"""Script with functions for driver node selections."""
+
+from typing import Any, List, Set
+
+from network_diffusion.mln.actor import MLNetworkActor
+from network_diffusion.mln.mlnetwork import MultilayerNetwork
+
+
+def compute_driver_nodes(net: MultilayerNetwork) -> List[MLNetworkActor]:
+    """Return driver nodes for a given network."""
+    min_dominating_set: Set[Any] = set()
+    for layer in net.layers:
+        min_dominating_set = minimum_dominating_set_with_initial(
+            net, layer, min_dominating_set
+        )
+
+    return [net.get_actor(actor_id) for actor_id in min_dominating_set]
+
+
+def minimum_dominating_set_with_initial(
```

Comment:
Please, add a reference in docstring to a paper, where you got this routine from. And if you, yourself, invented this algorithm, then I'm very impressed, and we should mention that in the paper 😄

Reply:
I extended the well-known MDS heuristic for multilayer networks. The original method can be found at https://link.springer.com/article/10.1007/s00453-001-0101-z  (Section 2.1, Greedy).  To align nodes across layers, I introduced an initial set and compared actor dominance.